### PR TITLE
java: Allow `enum { ... }` patterns

### PR DESCRIPTION
**Pull request by IagoAbal**
Fixes: semgrep#2715

test plan:
make test

### CODE REVIEW

**Comment by aryx** [comment]
you need to setup your pre-commit hook for pfff too. See the CI failure.

**Review by aryx — APPROVED** [review]

File: tests/java/parsing/EnumEllipsis.java [hunk 1]
```diff
@@ -0,0 +1,2 @@
+enum $A { ... }
```
Comment (aryx) [inline tests/java/parsing/EnumEllipsis.java:1]: this is not valid Java code, it should not be in tests/java/parsing/

Comment (IagoAbal) [reply tests/java/parsing/EnumEllipsis.java:1]: Is there a good place to put in Pfff? Otherwise I can remove the test and add the test in Semgrep instead.

Comment (aryx) [reply tests/java/parsing/EnumEllipsis.java:1]: Yep I did that. For some languages I had a pfff/tests/<lang>/sgrep/ subdirectory to put those tests, but now it's better to have them in semgrep I think.

**Pull request merged**